instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/messaging/MessagingSpanNameExtractor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.messaging;
+
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+public final class MessagingSpanNameExtractor<REQUEST> implements SpanNameExtractor<REQUEST> {
+
+  /**
+   * Returns a {@link SpanNameExtractor} that constructs the span name according to <a
+   * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#span-name>
+   * messaging semantic conventions</a>: {@code <destination name> <operation name>}.
+   *
+   * @see MessagingAttributesExtractor#destination(Object) used to extract {@code <destination
+   *     name>}.
+   * @see MessagingAttributesExtractor#operation(Object) used to extract {@code <operation name>}.
+   */
+  public static <REQUEST> SpanNameExtractor<REQUEST> create(
```

Comment:
Why does this class have a factory method, while `MessagingAttributesExtractor` does not? What is so different about this class that requires special construction?

Reply:
`MessagingAttributesExtractor` is an abstract class that has to be extended, it cannot have a static factory method.

> What is so different about this class that requires special construction?

Hmm, there's nothing special - I've added it here because we use builders & factory methods (instead of plain constructor calls) in the instrumenter API. I guess that's more resistant to future changes? But I may be overthinking this, and maybe it just looks a bit cooler 😄